projects/driver_controls/src/mechanical_brake_fsm.c

```diff
@@ -0,0 +1,84 @@
+#include "mechanical_brake_fsm.h"
+#include "input_event.h"
+#include "event_arbiter.h"
+#include "log.h"
+
+// Mechanical Brake FSM state definitions
+
+FSM_DECLARE_STATE(state_engaged);
+FSM_DECLARE_STATE(state_disengaged);
+
+// Mechanical Brake FSM transition table definitions
+
+FSM_STATE_TRANSITION(state_engaged) {
+  FSM_ADD_TRANSITION(INPUT_EVENT_MECHANICAL_BRAKE_PRESSED, state_engaged);
+  FSM_ADD_TRANSITION(INPUT_EVENT_MECHANICAL_BRAKE_RELEASED, state_disengaged);
+}
+
+FSM_STATE_TRANSITION(state_disengaged) {
+  FSM_ADD_TRANSITION(INPUT_EVENT_MECHANICAL_BRAKE_PRESSED, state_engaged);
+  FSM_ADD_TRANSITION(INPUT_EVENT_MECHANICAL_BRAKE_RELEASED, state_disengaged);
+}
+
+// Mechanical Brake FSM arbiter functions
+
+static bool prv_check_mechanical_brake_engaged(const Event *e) {
```

Comment:
Would definitely appreciate explanations for the event filters (why they're disabled in this state and how it fits into the general concept)